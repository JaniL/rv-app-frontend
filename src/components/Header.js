import React from 'react';
import DangerBtn from './DangerBtn';
import SuccessBtn from './SuccessBtn';

export class Header extends React.Component {
    constructor(props) {
        super(props);
        this.state = {};
    }

    render() {
        return (
            <div style={styles.header}>
              <div style={styles.selaa}>
                <SuccessBtn>Selaa valikoimaa</SuccessBtn>
              </div>
              <div style={styles.logout}>
                <DangerBtn>Kirjaudu ulos (ENTER)</DangerBtn >
              </div>
            </div>
        )
    }
}

const styles = {
    header: {
        width: '100%',
        height: '5%',
        display: 'inline-block'
    },
    selaa: {
        paddingTop: 5,
        float: 'left',
    },
    logout: {
        paddingTop: 5,
        paddingRight: 5,
        float: 'right',
    },
}