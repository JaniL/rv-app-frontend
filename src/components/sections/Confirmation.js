import React, { Component } from 'react';
import './styles/Confirmation.css';

export class Confirmation extends Component {
    render() {
        return (
            <div class="confirm-overlay">
                <div class="confirm">
                    Vahvista{' '}
                    <b>
                        {parseFloat(this.props.depositAmount / 100).toFixed(2)}{' '}
                        &euro;
                    </b>{' '}
                    deposit
                    <br />
                    <br />
                    <button
                        class="btn number cancel cancelBtn"
                        onClick={() => {
                            this.props.toggleConfirmationVisibility(true);
                        }}
                    >
                        Peruuta
                    </button>
                    <button
                        class="btn number success confirmBtn"
                        onClick={() => {
                            this.props.increaseBalance(
                                this.props.token,
                                this.props.depositAmount
                            );
                            this.props.resetAmount();
                            this.props.toggleConfirmationVisibility(true);
                            this.props.toggleVisibility(true);
                        }}
                    >
                        OK
                    </button>
                </div>
            </div>
        );
    }
}

export default Confirmation;
