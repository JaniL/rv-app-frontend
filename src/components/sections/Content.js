import React from 'react';
import './styles/Content.css';
import { Grid, Col, Row } from 'react-flexbox-grid';
import FeaturedProducts from './FeaturedProducts';
import ShoppingCart from './ShoppingCart';
import Terminal from './Terminal';

import { connect } from 'react-redux';

import {
    successMessage,
    errorMessage
} from './../../reducers/notificationReducer';

export class Content extends React.Component {
    render() {
        return (
            <main>
                <button
                    onClick={() =>
                        this.props.successMessage('Success message test')
                    }
                >
                    Trigger success
                </button>
                <button
                    onClick={() =>
                        this.props.errorMessage('Error message test')
                    }
                >
                    Trigger error
                </button>
                <Grid style={{width: '100%'}}>
                    <Row>
                        <Col xs={6}>
                            <FeaturedProducts />
                        </Col>
                        <Col xs={6}>
                            <ShoppingCart />
                        </Col>
                    </Row>
                    <Row />
                    <Row>
                        <Terminal deposit={this.props.deposit} />
                    </Row>
                </Grid>
            </main>
        );
    }
}

const mapDispatchToProps = {
    successMessage,
    errorMessage
};

const mapStateToProps = state => {
    return {
        success: state.notification.success,
        error: state.notification.error
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(Content);
