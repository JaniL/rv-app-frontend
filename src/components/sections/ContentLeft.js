import React from 'react';

export class ContentLeft extends React.Component {
    render() {
        return (
            <div style={styles.content}>
            </div>
        )
    }
}

const styles = {
    content: {
        height: '100%',
        width: '50%',
        float: 'left'
    }
}