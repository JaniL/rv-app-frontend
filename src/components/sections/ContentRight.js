import React from 'react';

export class ContentRight extends React.Component {
    render() {
        return (
            <div style={styles.content}>
            </div>
        )
    }
}

const styles = {
    content: {
        height: '100%',
        width: '50%',
        float: 'right'
    }
}