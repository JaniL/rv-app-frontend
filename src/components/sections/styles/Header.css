header {
  padding-top: 4px;
  padding-bottom: 4px;
  width: 100%;
  vertical-align: middle;
  box-sizing: border-box;
  -webkit-box-sizing: border-box;
  display: inline-block;
  box-shadow: 0 1px 2px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
}

.header-left {
  width: 50%;
  float: left;
  text-align: left;
}

.header-right {
  width: 50%;
  float: right;
  text-align: right;
}
