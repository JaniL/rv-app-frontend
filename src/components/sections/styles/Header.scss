@import 'vars';

header {
  padding-top: 4px;
  padding-bottom: 4px;
  width: 100%;
  vertical-align: middle;
  box-sizing: border-box;
  -webkit-box-sizing: border-box;
  background-color: $header-bgcolor;
  box-shadow: $box-shadow;
  display: flex;
  flex-direction: row;
  align-items: center;

  h1 {
    font-size: 1.6em;
    font-weight: normal;
    margin: 0 16px;
  }
}

.header-right {
  margin-left: auto;
}
