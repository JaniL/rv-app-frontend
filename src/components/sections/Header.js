import React from "react";
import DangerBtn from "../buttons/DangerBtn";
import SuccessBtn from "../buttons/SuccessBtn";
import BasicBtn from "../buttons/BasicBtn";
import Margin from "../helpers/Margin";

import "./styles/Header.css";

export class Header extends React.Component {
  render() {
    return (
      <header>
        <div className="header-left">
          <Margin margin={5} inlineBlock>
            <BasicBtn
              hover
              onClick={() =>
                this.props.buy({
                  price: 180,
                  name: "Coca-Cola Zero 0.5l"
                })
              }
            >
              Testiostos (1.8 &euro;)
            </BasicBtn>
          </Margin>
          <Margin margin={5} inlineBlock>
            <BasicBtn
              hover
              onClick={() =>
                this.props.store({
                  price: 500,
                  name: "Talletus"
                })
              }
            >
              Testitalletus (5 &euro;)
            </BasicBtn>
          </Margin>
        </div>
        <div className="header-right">
          <Margin margin={5} inlineBlock>
            <span>
              <b>{this.props.user.full_name}</b>{" "}
              {this.props.user.account_balance / 100} &euro;
            </span>
          </Margin>
          <Margin margin={5} inlineBlock>
            <SuccessBtn hover>Asetukset</SuccessBtn>
          </Margin>
          <Margin margin={5} inlineBlock>
            <DangerBtn onClick={this.props.logout} hover>
              Kirjaudu ulos (ENTER)
            </DangerBtn>
          </Margin>
        </div>
      </header>
    );
  }
}
