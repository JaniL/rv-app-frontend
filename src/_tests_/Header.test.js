import React from 'react';
import ReactDOM from 'react-dom';
import { Header } from '../components/sections/Header';
import Enzyme from 'enzyme';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Header />, div);
});