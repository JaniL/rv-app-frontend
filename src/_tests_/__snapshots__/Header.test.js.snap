// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<header>
  <div
    className="header-left"
  >
    <div
      className="margin"
      style={
        Object {
          "display": "inline-block",
          "margin": 5,
        }
      }
    >
      <button
        className="btn basic basic-hover"
        onClick={[Function]}
      >
        <span
          className="btnContent"
        >
          Testiostos 1 (
          1.80
           
          €)
        </span>
      </button>
    </div>
    <div
      className="margin"
      style={
        Object {
          "display": "inline-block",
          "margin": 5,
        }
      }
    >
      <button
        className="btn basic basic-hover"
        onClick={[Function]}
      >
        <span
          className="btnContent"
        >
          Testiostos 2 (
          1.20
           
          €)
        </span>
      </button>
    </div>
    <div
      className="margin"
      style={
        Object {
          "display": "inline-block",
          "margin": 5,
        }
      }
    >
      <button
        className="btn basic basic-hover"
        onClick={[Function]}
      >
        <span
          className="btnContent"
        >
          Testitalletus (
          5.00
           
          €)
        </span>
      </button>
    </div>
  </div>
  <div
    className="header-right"
  >
    <div
      className="margin"
      style={
        Object {
          "display": "inline-block",
          "margin": 5,
        }
      }
    >
      <button
        className="btn basic"
        onClick={null}
      >
        <span
          className="btnContent"
        >
          <span>
            <b>
              John Doe
            </b>
             
            10.00
             
            €
          </span>
        </span>
      </button>
    </div>
    <div
      className="margin"
      style={
        Object {
          "display": "inline-block",
          "margin": 5,
        }
      }
    >
      <button
        className="btn danger danger-hover"
        onClick={null}
      >
        <span
          className="btnContent"
        >
          Kirjaudu ulos (ENTER)
        </span>
      </button>
    </div>
  </div>
</header>
`;
